lms/djangoapps/courseware/grades.py

```diff
@@ -737,24 +746,17 @@ def get_score(user, problem_descriptor, module_creator, scores_client, submissio
     else:
         # This means we don't have a valid score entry and we don't have a
         # cached_max_score on hand. We know they've earned 0.0 points on this,
-        # but we need to instantiate the module (i.e. load student state) in
-        # order to find out how much it was worth.
-        problem = module_creator(problem_descriptor)
-        if problem is None:
-            return (None, None)
-
         correct = 0.0
-        total = problem.max_score()
+        total = block.transformer_data[GradesTransformer].max_score
 
-        # Problem may be an error module (if something in the problem builder failed)
-        # In which case total might be None
+        # Handle any edge cases
```

Comment:
hmm.. the reason listed in the previous comment is no longer valid since the block instantiation now happens in the collect phase and the `total` and `weight` fields are always cached.   So I removed checking for this case.